apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/controller/UserInfoControllerTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.portal.controller;
+
+import com.ctrip.framework.apollo.common.exception.BadRequestException;
+import com.ctrip.framework.apollo.portal.entity.po.UserPO;
+import com.ctrip.framework.apollo.portal.spi.springsecurity.SpringSecurityUserService;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserInfoControllerTest {
+
+  @Mock
+  private UserInfoController userInfoController;
+  @InjectMocks
+  private SpringSecurityUserService userService;
+
+  @Test
+  public void testCreateOrUpdateUser() {
+    UserPO user = new UserPO();
+    user.setUsername("username");
+    user.setPassword("af12gvi3ox");
+    userInfoController.createOrUpdateUser(user);
+  }
+
+  @Test
+  public void testCreateOrUpdateUserFailed() {
+    List<String> pwdCases = Arrays.asList("1111111", "1111qwer");
+    for (String p : pwdCases) {
+      UserPO user = new UserPO();
+      user.setUsername("username");
+      user.setPassword(p);
+      try {
+        userInfoController.createOrUpdateUser(user);
+      } catch (BadRequestException e) {
+        Assert.assertEquals(
```

Comment:
It's better we mock password checker here instead of testing the password logic here, as it is not core logic of UserInfoController.

Reply:
Ok, done.